mongoimport/options_test.go

```diff
@@ -100,3 +101,135 @@ func TestLegacyOptionParsing(t *testing.T) {
 		})
 	}
 }
+
+type PositionalArgumentTestCase struct {
+	InputArgs    []string
+	ExpectedOpts Options
+	ExpectErr    string
+}
+
+func TestPositionalArgumentParsing(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.UnitTestType)
+	Convey("Testing parsing positional arguments", t, func() {
+		positionalArgumentTestCases := []PositionalArgumentTestCase{
+			{
+				InputArgs: []string{"foo"},
+				ExpectedOpts: Options{
+					ToolOptions: &options.ToolOptions{
+						URI: &options.URI{
+							ConnectionString: "mongodb://localhost/",
+						},
+					},
+					InputOptions: &InputOptions{
+						File: "foo",
+					},
+				},
+			},
+			{
+				InputArgs: []string{"mongodb://foo"},
+				ExpectedOpts: Options{
+					ToolOptions: &options.ToolOptions{
+						URI: &options.URI{
+							ConnectionString: "mongodb://foo",
+						},
+					},
+					InputOptions: &InputOptions{},
+				},
+			},
+			{
+				InputArgs: []string{"mongodb://foo", "foo"},
+				ExpectedOpts: Options{
+					ToolOptions: &options.ToolOptions{
+						URI: &options.URI{
+							ConnectionString: "mongodb://foo",
+						},
+					},
+					InputOptions: &InputOptions{
+						File: "foo",
+					},
+				},
+			},
+			{
+				InputArgs: []string{"foo", "mongodb://foo"},
+				ExpectedOpts: Options{
+					ToolOptions: &options.ToolOptions{
+						URI: &options.URI{
+							ConnectionString: "mongodb://foo",
+						},
+					},
+					InputOptions: &InputOptions{
+						File: "foo",
+					},
+				},
+			},
+			{
+				InputArgs: []string{"foo", "--uri=mongodb://foo"},
+				ExpectedOpts: Options{
+					ToolOptions: &options.ToolOptions{
+						URI: &options.URI{
+							ConnectionString: "mongodb://foo",
+						},
+					},
+					InputOptions: &InputOptions{
+						File: "foo",
+					},
+				},
+			},
+			{
+				InputArgs: []string{"--file=foo", "mongodb://foo"},
+				ExpectedOpts: Options{
+					ToolOptions: &options.ToolOptions{
+						URI: &options.URI{
+							ConnectionString: "mongodb://foo",
+						},
+					},
+					InputOptions: &InputOptions{
+						File: "foo",
+					},
+				},
+			},
+			{
+				InputArgs: []string{"mongodb://foo", "mongodb://bar"},
+				ExpectErr: "too many URIs found in positional arguments: only one URI can be set as a positional argument",
+			},
+			{
+				InputArgs: []string{"foo", "bar"},
+				ExpectErr: "error parsing positional arguments: " +
+					"provide only one polling interval in seconds and only one MongoDB connection string. " +
+					"Connection strings must begin with mongodb:// or mongodb+srv:// schemes",
+			},
+			{
+				InputArgs: []string{"foo", "bar", "mongodb://foo"},
+				ExpectErr: "error parsing positional arguments: " +
+					"provide only one polling interval in seconds and only one MongoDB connection string. " +
```

Comment:
Looks like this might be a copy-and-pase err from another tool's tests?

Reply:
Good catch!